torch/lib/c10d/ProcessGroupMPI.hpp

```diff
@@ -42,7 +42,11 @@ struct WorkEntry {
 
   // For input and output tensors (in-place), we will always use src
   std::vector<at::Tensor> src;
-  std::vector<at::Tensor> dst;
+
+  // We use `dst` for returning resulting vectors in WorkMPI::result, so we need to
+  // keep it alive.
+  std::shared_ptr<std::vector<at::Tensor>> dst;
```

Comment:
could you please elaborate why the original `std::vector<at::Tensor>` was not sufficient? Is it because you wanna avoid making a copy of the vector when retrieving the outputs?

Reply:
this issue was for vectors of tensors.